### ISSUE TITLE
Request to duplicate transactions

### ISSUE THREAD

**Issue by triplejranch** [actualbudget/releases#220]
I searched and didn't find this, but forgive me if I missed it.  It would be really handy if we could select a transaction and duplicate it so we could just change the date and be on our way.  I'm thinking of transactions I have that have multiple splits and I repeat monthly.  A big time saver.

**Comment by Deguello** [comment]
For non-splits, you can set up schedules. For splits, I think that ability is being added to schedule functionality.

I agree it would be nice to copy and paste a transaction. Copy and paste with a transaction selected seems to do nothing. If it's not going to do anything, there should be an indiction of that in the menu. Right now, Copy looks greyed out, but it copies individual fields in a transaction but appears to copy nothing if a transaction is selected.

**Comment by triplejranch** [comment]
Yes, if schedules could do splits, then I'd be set.  Copy and paste hasn't worked, I agree.  Just thought being able to duplicate might be an option.  Thanks:)

**Comment by Deguello** [comment]
Honestly, for now, I just broke my splits into separate transactions. First, it made scheduling easier. Second, the way drill downs from the budget screen work right now, amounts in splits don't show up right and this avoids the issue. Third, they all happen at once, anyway, so clearing them all at once works OK.

I agree with your suggestion and I'm not saying this is how it SHOULD work, but doing it this way is better than nothing for me.

**Comment by Kidglove57** [comment]
_“I just broke my splits into separate transactions”_
 
Same here and so far all has been fine. It seemed a big deal initially but not so much when I adapted to it. I add a note to indicate that the transaction is recorded in two parts, something like: 
1 of 2 total £60
 and then 
2 of 2 total £60.
But yes I agree it’s not ideal

**Comment by triplejranch** [comment]
Ok.  To clarify, you're adding separate transactions, not separate schedules? Well, I guess it doesn't matter.  Gah, that seems like so much extra work, but then again, I guess it's pretty much the same. Sorry, talking out loud, lol.   If I made separate schedule for all my splits, then I would only be doing it once.  Got it.  Thanks for the workaround tip all!  Hey, do you happen to know where this is on his roadmap before I dive into this?

**Comment by triplejranch** [comment]
Just learned about the tool "Find Schedules".  So handy.  Never mind my last comment.

**Comment by Kidglove57** [comment]
I do have a couple of schedules split into separate transactions. No it's not ideal -  but it does the job in the short term whilst this feature is being worked on.

**Comment by triplejranch** [comment]
Hey, I can't get Tools: Find Schedules to do anything.  Just nothing. Is there a trick I'm doing wrong or do I not have enough data to trigger anything? I have  3 months worth of transactions, but maybe that's not enough?  I thought it would make it easier fo rme, but I can't get it to work.

**Comment by jlongster** [comment]
@triplejranch hey! it's hard to say -- it will look for something occurring 3 times I believe. So you may not have just enough data yet. They also have the match payees.

Duplicating is an interesting idea, but I'd rather invest time in getting splits to work in rules (and thus schedules) :) I think I'm gonna work on this very soon, not the next release but the one after that. I'm working on https://github.com/actualbudget/releases/issues/218 so it would be natural to allow splits in rules after this, then you can have scheduled transfers in splits as well.
